packages/aws-cdk-lib/aws-lambda/lib/adot-layers.ts

```diff
@@ -184,6 +184,11 @@ export enum AdotLambdaExecWrapper {
    * propagation for HTTP requests.
    */
   STREAM_HANDLER = '/opt/otel-stream-handler',
+
+  /**
+   * Wrapping python lambda handlers see https://aws-otel.github.io/docs/getting-started/lambda/lambda-python
+   */
+  INSTRUMENT_HANDLER = '/opt/otel-instrument',
```

Comment:
This is so unfortunate 😒.

From the documentation it looks like `/opt/otel-instrument` is the only possible
value for python? If that is the case let just not even add an option and set
the value here based on the runtime.

https://github.com/aws/aws-cdk/blob/53d24f1a1357410746052bbf25b5b89b985f7b80/packages/aws-cdk-lib/aws-lambda/lib/function.ts#L1161-L1161

Reply:
Not sure I follow on this one. The `this._layers.map` could be any number of layers even beyond the adot ones. What exactly would I filter on? I could maybe validate in `configureAdotInstrumentation` check for the `AdotLayerVersion` and if it's python set the `AWS_LAMBDA_EXEC_WRAPPER` env variable to `opt/otel-instrument`. Is that what you were thinking? I like either forcing the env variable or throwing an error if they select the wrong handler.

Reply:
In the `configureAdotInstrumentation` function I could do something like below to check the layer and throw and error if it's not `INSTRUMENT_HANDLER`. I would like to keep ADOT work separate from the generic lambda layer/env work. Or instead of throwing the error I could swap to the instrument handler but I worry about an end user setting a value and us swapping it out and what ramifications it could have in the future. The throwing of the error would be clear and it's a one time change.

```
    if (props.adotInstrumentation.layerVersion instanceof AdotLambdaLayerPythonSdkVersion
      && props.adotInstrumentation.execWrapper != AdotLambdaExecWrapper.INSTRUMENT_HANDLER) {
      throw new Error('Python Adot Lambda layer requires AdotLambdaExecWrapper.INSTRUMENT_HANDLER');
    }
```
Thoughts?

Reply:
I decided to go with throwing an error when a python function is using the Adot layer when not using the EXEC_WRAPPER INSTRUMENT_HANDLER. I wanted to keep the logic for all adot within the configuration function `configureAdotInstrumentation` to isolate that work. This seemed like the best approach going forward. This last commit is ready for review.